### PR TITLE
[jasmine-reporters] Add types

### PR DESCRIPTION
**Pull request by akwiatek**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/jasmine-reporters/tslint.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "extends": "@definitelytyped/dtslint/dt.json",
+    "rules": {
+        // Classes are used so they can implment Jasmine interfaces.
+        "no-unnecessary-class": false
```
Comment (akwiatek) [inline types/jasmine-reporters/tslint.json:5]: The rule complained about e.g. `export class AppVeyorReporter implements jasmine.CustomReporter {`

For documentation purposes I have included the Jasmine interface. AFAIK I should have a class to implement an interface.

According to the documentation of this library officially there are no additional methods / fields exposed in constructed objects.
Everything we know about those objects is that they are compatible with Jasmine.
See: https://github.com/larrymyers/jasmine-reporters#basic-usage

File: types/jasmine-reporters/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
+// Type definitions for Jasmine Reporters 2.5
+// Project: https://github.com/larrymyers/jasmine-reporters
+// Definitions by: Adam Kwiatek <https://github.com/akwiatek>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import jasmine = require('jasmine');
+
+// tslint:disable-next-line:no-unnecessary-class
+export class AppVeyorReporter implements jasmine.CustomReporter {
```
Comment (akwiatek) [inline types/jasmine-reporters/index.d.ts:9]: I have added classes so they could implement `jasmine.CustomReporter`.

File: types/jasmine-reporters/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    constructor(options?: AppVeyorReporter.ConstructorOptions);
+}
+
+export namespace AppVeyorReporter {
+    interface ConstructorOptions {
+        batchSize: number;
+        color: boolean;
+        verbosity: 0 | 1 | 2;
```
Comment (peterblazejewicz) [inline types/jasmine-reporters/index.d.ts:17]: all of thos are marked as optional in constructor jsdoc:
https://github.com/larrymyers/jasmine-reporters/blob/master/src/appveyor_reporter.js#L32-L35

Comment (akwiatek) [reply types/jasmine-reporters/index.d.ts]: Done.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@akwiatek thanks!

**Comment by akwiatek** [comment]
Ready to merge

**Pull request merged**